components/shared/LeftSidebar/CommandPalette/CommandPalette.tsx

```diff
@@ -0,0 +1,130 @@
+"use client";
+import React, { useEffect, MouseEvent } from "react";
+import { Command } from "cmdk";
+import Image from "next/image";
+import { postFilters, testTags } from "~/constants";
+import { useRouter } from "next/navigation";
+
+interface CommandPaletteProps {
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+}
+
+const CommandPalette: React.FC<CommandPaletteProps> = ({
+  isOpen,
+  setIsOpen,
+}) => {
+  const router = useRouter();
+
+  useEffect(() => {
+    const down = (e: any) => {
+      if (e.key === "k" && (e.metaKey || e.ctrlKey)) {
+        e.preventDefault();
+        setIsOpen(!isOpen);
+      }
+      //check if the key pressed is the escape key
+      if (e.key === "Escape") {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener("keydown", down);
+    return () => document.removeEventListener("keydown", down);
+  }, [isOpen, setIsOpen]);
+
+  const handleOverlayClick = (event: MouseEvent<HTMLDivElement>) => {
+    event.stopPropagation();
+    setIsOpen(!isOpen);
+  };
+
+  return (
+    <div>
+      {isOpen && (
+        <div className="cmdk-overlay" onClick={handleOverlayClick}>
+          <div className="cmdk-modal" onClick={(e) => e.stopPropagation()}>
+            <Command className="bg-myBlack-900">
+              <div className="flex w-full justify-between gap-2 bg-myBlack-700 p-2 text-myWhite-300">
+                <div className="flex">
+                  <Image
+                    src="/search-icon.svg"
+                    alt="Search"
+                    width={12}
+                    height={12}
+                  ></Image>
+                  <Command.Input
+                    className="w-full bg-myBlack-700 p-2 text-myWhite-300"
+                    placeholder="Search..."
+                    autoFocus={true}
+                  />
+                </div>
+                <button
+                  className="flex items-center justify-center rounded-sm bg-myBlack-800 p-1 text-p4Reg text-myWhite-300 hover:bg-myBlack-600 hover:text-myWhite-100"
+                  onClick={() => setIsOpen(false)}
+                >
+                  {`ESC`}
+                </button>
+              </div>
+
+              <Command.List className="flex flex-col bg-myBlack-800">
```

Comment:
```suggestion
              <Command.List className="flex h-[200px] flex-col overflow-y-scroll bg-myBlack-800">
```

You might a more reasonable time with the design if we set some max height- and force the scroll to happen inside the results instead of inside the entire modal. Play with the height and see what a good value would be.

![image](https://github.com/dalbanhi/git-note/assets/4108484/51b521fd-0492-499e-91b8-278e6057b471)

You could even look into customizing the scrollbar a bit by adding a "customScroll" class here and then adding something into our globals:

```js
.customScroll {
  scrollbar-width: thin;
  scrollbar-color: #4a4e69 #1d2032;
}

```
![image](https://github.com/dalbanhi/git-note/assets/4108484/29491fd4-df77-4966-8dcc-2148f589acbc)


This is just an example- but experiment here to get a quality looking component and try using the design for reference here :)